components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -1659,6 +1765,167 @@ public Response apisHead(String query, String ifNoneMatch, MessageContext messag
+                            + "importing schema: " + schema);
+                }
+            }
+
+            additionalPropertiesAPI = new ObjectMapper().readValue(additionalProperties, APIDTO.class);
+            additionalPropertiesAPI.setType(APIDTO.TypeEnum.GRAPHQL);
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            API apiToAdd = prepareToCreateAPIByDTO(additionalPropertiesAPI);
+            //adding the api
+            apiProvider.addAPI(apiToAdd);
+
+            //Save swagger definition of graphQL
+            APIDefinitionFromOpenAPISpec apiDefinitionUsingOASParser = new APIDefinitionFromOpenAPISpec();
+            String apiDefinition = apiDefinitionUsingOASParser.generateAPIDefinition(apiToAdd);
+            apiProvider.saveSwagger20Definition(apiToAdd.getId(), apiDefinition);
+
+            APIIdentifier createdApiId = apiToAdd.getId();
+
+            //Retrieve the newly added API to send in the response payload
+            API createdApi = apiProvider.getAPI(createdApiId);
+
+            String resourcePath = createdApiId.getProviderName() + APIConstants.GRAPHQL_SCHEMA_PROVIDER_SEPERATOR +
+                    createdApiId.getApiName() + createdApiId.getVersion() +
+                    APIConstants.GRAPHQL_SCHEMA_FILE_EXTENSION;
+            resourcePath = APIConstants.API_GRAPHQL_SCHEMA_RESOURCE_LOCATION + resourcePath;
+
+            if (apiProvider.checkIfResourceExists(resourcePath)) {
+                RestApiUtil.handleConflict("schema resource already exists for the API " +
+                        additionalPropertiesAPI.getId(), log);
+            }
+            apiProvider.uploadGraphqlSchema(resourcePath, schema);
+            APIDTO createdApiDTO = APIMappingUtil.fromAPItoDTO(createdApi);
+
+            //This URI used to set the location header of the POST response
+            URI createdApiUri = new URI(RestApiConstants.RESOURCE_PATH_APIS + "/" + createdApiDTO.getId());
+            return Response.created(createdApiUri).entity(createdApiDTO).build();
+
+        } catch (APIManagementException e) {
+            String errorMessage = "Error while adding new API : " + additionalPropertiesAPI.getProvider() + "-" +
+                additionalPropertiesAPI.getName() + "-" + additionalPropertiesAPI.getVersion() + " - " + e.getMessage();
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        } catch (URISyntaxException e) {
+            String errorMessage = "Error while retrieving API location : " + additionalPropertiesAPI.getProvider() + "-" +
+                additionalPropertiesAPI.getName() + "-" + additionalPropertiesAPI.getVersion();
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+    } catch (IOException e) {
+            String errorMessage = "Error while retrieving content from file : " + additionalPropertiesAPI.getProvider()
```

Comment:
check whether line 1835, 1838 and 1842 exceeds 120 line break

Reply:
Fixed with 2a6fbdd3ac9ce54ca7bc94a45328a41826836a80